src/app/clusters/fan-control-server/fan-control-server.cpp

```diff
@@ -298,6 +312,40 @@ MatterFanControlClusterServerPreAttributeChangedCallback(const ConcreteAttribute
         }
         break;
     }
+    case PercentCurrent::Id: {
+        // Check if the PercentSetting is null.
+        if (NumericAttributeTraits<Percent>::IsNullValue(*value))
+        {
+            // nothing to do?
+            res = Status::Success;
+        }
+        else
+        {
+            // default to suppressing a report
+            res = Status::SuppressReport;
+            if (percentCurrentQTracking[ep].moveStarted)
+            {
+                // need a report at the start of a move.
+                percentCurrentQTracking[ep].moveStarted = false;
+                percentCurrentQTracking[ep].lastUpdateTime = GetTimeNowSomehow();
```

Comment:
If we're going to store this stuff out-of-band like this anyway, it's probably a good idea to use src/app/cluster-building-blocks/QuieterReporting.h bits to store a bunch of this state.